### ISSUE TITLE
[Feature] Add callback support for ExecutionMode.KUBERNETES

### ISSUE THREAD

**Issue by pankajkoti** [astronomer/astronomer-cosmos#1398]
### Description

Similar to the functionality of supporting to specify callback argument in the ExecutionMode.LOCAL as can be seen in https://github.com/astronomer/astronomer-cosmos/blob/0000f80cbdb1e137b423a9297d45380441dd8c9b/cosmos/operators/local.py#L143, and as PR #1389 demonstrates how to leverage this callback argument, we should add the support for the callback argument in `ExceutionMode.KUBERNETES`

### Use case/motivation

A common request from users is the ability to inspect files in the target directory after task execution. Since Cosmos clones the project directory into a temporary directory and executes tasks from there, access to this directory is lost once the task completes. While ExecutionMode.LOCAL provides callback support to perform actions on this temporary directory, similar functionality is currently unavailable for ExecutionMode.Kubernetes. Adding this support would be highly beneficial.

### Related issues

related: #1349 

### Are you willing to submit a PR?

- [ ] Yes, I am willing to submit a PR!

**Comment by pankajkoti** [comment]
@jroach-astronomer, the issue stems from Cosmos cloning the project directory into a temporary location and executing tasks from there. Once the task completes, access to this directory is lost because the system often deletes the temporary directory(often immediately). As a result, any files written to that temporary directory during execution are no longer accessible.

**Comment by jroach-astronomer** [comment]
For this execution mode. do we spin up a new Pod for each Task run? Maybe a shared volume for an init container and a base container?

**Comment by pankajkoti** [comment]
Yes, a new pod is created, but the key point remains that the temporary directory’s cleanup depends on the system, storage type, and configured policies, which can still impact access to the artefacts.

The goal is to introduce a callback mechanism for ExecutionMode.KUBERNETES, allowing a function to be specified that uploads files from the temporary directory to a remote blob store during the execution of the dbt command—before the temporary directory is removed. This functionality is already supported in ExecutionMode.LOCAL, as mentioned in the issue description, and we aim to extend it to Kubernetes mode.

**Comment by tatiana** [comment]
Closing this as it is a duplicate of https://github.com/astronomer/astronomer-cosmos/issues/1575
